pysignalscope/scope.py

```diff
@@ -949,86 +1007,79 @@ def short_to_period(self, f0: Union[float, int, None] = None, time_period: Union
 
         Returns: rms(self.channel_data).
         """
         # Log flow control
-        logging.debug(f"{self.modulename} :Number of channel data={len(self.channel_data)}")
+        logging.debug(f"{channel.modulename} :Number of channel data={len(channel.channel_data)}")
 
-        return np.sqrt(np.mean(self.channel_data ** 2))
+        return np.sqrt(np.mean(channel.channel_data ** 2))
 
-    def mean(self) -> Any:
+    @staticmethod
+    def mean(channel: Scope) -> Any:
         """
         Calculate the mean of the given channel. Make sure to provide a SINGLE PERIOD of the signal.
 
         Returns: mean(self.channel_data).
         """
         # Log flow control
-        logging.debug(f"{self.modulename} :Number of channel data={len(self.channel_data)}")
+        logging.debug(f"{channel.modulename} :Number of channel data={len(channel.channel_data)}")
 
-        return np.mean(self.channel_data)
+        return np.mean(channel.channel_data)
 
-    def absmean(self) -> Any:
+    @staticmethod
+    def absmean(channel: Scope) -> Any:
         """
         Calculate the absolute mean of the given channel. Make sure to provide a SINGLE PERIOD of the signal.
 
         Returns: abs(mean(self.channel_data)).
         """
         # Log flow control
-        logging.debug(f"{self.modulename} :Number of channel data={len(self.channel_data)}")
+        logging.debug(f"{channel.modulename} :Number of channel data={len(channel.channel_data)}")
 
-        return np.mean(np.abs(self.channel_data))
+        return np.mean(np.abs(channel.channel_data))
 
-    def abs(self) -> None:
+    @staticmethod
+    def abs(channel: Scope) -> Scope:
         """
         Modify the existing scope channel so that the signal is rectified.
 
-        Returns: abs(self.channel_data).
+        Returns: abs(channel.channel_data).
         """
         # Log flow control
-        logging.debug(f"{self.modulename} :Number of channel data={len(self.channel_data)}")
+        logging.debug(f"{channel.modulename} :Number of channel data={len(channel.channel_data)}")
 
-        self.channel_data = np.abs(self.channel_data)
-        if self.channel_label is not None:
-            self.channel_label = '|' + self.channel_label + '|'
+        channel.channel_data = np.abs(channel.channel_data)
+        if channel.channel_label is not None:
+            channel.channel_label = '|' + channel.channel_label + '|'
 
-    def square(self) -> None:
+        return channel
+
+    @staticmethod
+    def square(channel: Scope) -> Scope:
         """
-        Modify the existing scope channel so that the signal is squared.
+        Square the data channel.
 
-        Returns: self.channel_data ** 2.
+        Returns: channel.channel_data ** 2.
         """
-        self.channel_data = self.channel_data ** 2
-        if self.channel_label is not None:
-            self.channel_label = self.channel_label + '²'
+        channel.channel_data = channel.channel_data ** 2
```

Comment:
Before calculation of absolute vales a copy of the data are necessary. Otherwise you overwrite the origin channel data.

Reply:
fixed